scripts/core/editor3/html/to-html/BlockInlineStyleWrapper.js

```diff
@@ -87,9 +87,15 @@ export class BlockInlineStyleWrapper {
             return '';
         }
 
-        const noLongerApplied = this.activeStyles
+        let noLongerApplied = this.activeStyles
             .filter((s) => styles.toArray().indexOf(s) === -1);
 
+        if (noLongerApplied.length !== 0) {
+            const index = this.activeStyles.indexOf(noLongerApplied[0]);
```

Comment:
Could you explain how this works and what does it fix?

Reply:
Let's say that we have next `activeStyles` ["ITALIC", "BOLD", "UNDERLINED"] and next character is not <b>bold</b> anymore, next expression will return style which is no longer applied:
https://github.com/superdesk/superdesk-client-core/blob/9072076e47b7294c3bd5291e55e406e09877fb59/scripts/core/editor3/html/to-html/BlockInlineStyleWrapper.js#L90-L91
so in our case it's `"BOLD"`, after (return statement) based on this data we will return `</b>` which is wrong, because we need to close `</u>` (children) before `</b>`
That's why, we take index of no longer applied tag and take slice in reverse direction, because we need to close all child tags which are inside `noLongerApplied`

Reply:
@tomaskikutis try to log `noLongerApplied` variable before `return` and try to comment my fix, you will see the difference.